src/android/java/com/pspdfkit/cordova/action/annotation/RemoveAnnotationAction.java

```diff
@@ -51,9 +51,12 @@ protected void execAction(JSONArray args, CallbackContext callbackContext) throw
     final PdfDocument document = pdfActivity.getDocument();
     if (document != null) {
       AnnotationProvider annotationProvider = document.getAnnotationProvider();
-      annotationProvider.getAllAnnotationsOfType(getAnnotationTypeFromString(type), pageIndex, 1)
+      pdfActivity.addSubscription(annotationProvider.getAllAnnotationsOfType(
+          getAnnotationTypeFromString(type),
+          pageIndex,
+          1)
+          .observeOn(Schedulers.io())
           .filter(annotationToFilter -> !name.isEmpty() && name.equals(annotationToFilter.getName()))
-          .subscribeOn(Schedulers.io())
```

Comment:
Because getAllAnnotationsOfType has internal subscriber declared we use `observeOn` to switch for filtering.